modules/build/src/main/scala/scala/build/internal/markdown/MarkdownCodeBlockUtil.scala

```diff
@@ -0,0 +1,122 @@
+package scala.build.internal.markdown
+
+import scala.collection.mutable
+
+/** A util for handling of code blocks in Markdown. */
+object MarkdownCodeBlockUtil {
+  private val allowIndentedFence: Boolean = false
+
+  /** Representation for a code block contained in Markdown
+    *
+    * @param info
+    *   a list of tags tied to a given code block
+    * @param body
+    *   the code block content
+    * @param startLine
+    *   starting line on which the code block was defined
+    * @param endLine
+    *   end line on which the code block was closed
+    */
+  case class MarkdownFence(
+    info: Seq[String],
+    body: String,
+    startLine: Int, // start of fenced body EXCLUDING backticks
+    endLine: Int    // same as above
+  ) {
+    override def toString: String =
+      s"Fence[$info, lines $startLine-$endLine]{${body.replace("\n", "\\n")}}"
+
+    /** @return
+      *   `true` if this snippet should be ignored, `false` otherwise
+      */
+    def shouldIgnore: Boolean = info.head != "scala" || info.contains("ignore")
```

Comment:
There is any validation of info input? 
What happened if user pass?
```
scala test-reset raw
```

Reply:
Better error handling will be addressed in a separate PR

Reply:
IMO this one is large enough already 🤡